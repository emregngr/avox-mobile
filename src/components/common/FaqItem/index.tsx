import { MaterialCommunityIcons } from '@expo/vector-icons'
import { useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'
import Collapsible from 'react-native-collapsible'

import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { FaqItemType } from '@/types/common/faq'

interface FaqItemProps {
  index: number
  isExpanded: boolean
  item: FaqItemType
  toggleExpanded: (index: number) => void
}

export const FaqItem = ({ index, isExpanded, item, toggleExpanded }: FaqItemProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { description, title } = item
  return (
    <View className="mb-2">
      <TouchableOpacity
        activeOpacity={0.7}
        className="p-4 rounded-xl overflow-hidden bg-background-secondary"
        hitSlop={10}
        onPress={() => toggleExpanded(index)}
        testID={`faq-${item?.id}`}
      >
        <View className="flex-row justify-between items-center">
          <View className="flex-1 pr-4">
            <ThemedText color="text-100" type="body1">
              {title}
            </ThemedText>
          </View>
          <MaterialCommunityIcons
            color={colors?.onPrimary50}
            name={isExpanded ? 'chevron-up' : 'chevron-down'}
            size={24}
          />
        </View>
      </TouchableOpacity>

      <Collapsible collapsed={!isExpanded}>
        <View className="mt-2 p-4 rounded-xl overflow-hidden bg-background-tertiary">
          <ThemedText color="text-90" type="body2">
            {description}
          </ThemedText>
        </View>
      </Collapsible>
    </View>
  )
}
