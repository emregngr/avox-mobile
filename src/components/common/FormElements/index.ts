export * from './CheckboxField'
export * from './TextInputField'
