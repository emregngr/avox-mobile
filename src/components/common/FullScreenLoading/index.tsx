import React from 'react'
import { ActivityIndicator, View } from 'react-native'

export const FullScreenLoading = () => (
  <View
    className="flex-1 items-center justify-center bg-background-primary"
    testID="full-screen-loading"
  >
    <ActivityIndicator className="color-text-100" size="large" testID="loading-indicator" />
  </View>
)
