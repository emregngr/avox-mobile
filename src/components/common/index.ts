export * from './FaqItem'
export * from './FormElements'
export * from './FullScreenLoading'
export * from './Header'
export * from './OnboardingItem'
export * from './ProfileItem'
export * from './RenderDetailTabBar'
export * from './RenderTabBar'
export * from './SafeLayout'
export * from './SearchInput'
export * from './ThemedButton'
export * from './ThemedButtonText'
export * from './ThemedGradientButton'
export * from './ThemedText'
export * from './ToastConfig'
