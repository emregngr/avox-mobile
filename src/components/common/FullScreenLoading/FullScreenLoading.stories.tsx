import type { Meta, StoryObj } from '@storybook/react-native'
import React, { useState } from 'react'
import { TouchableOpacity, View } from 'react-native'

import { FullScreenLoading } from '@/components/common/FullScreenLoading'
import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

const meta: Meta<typeof FullScreenLoading> = {
  title: 'Components/FullScreenLoading',
  component: FullScreenLoading,
  parameters: {
    docs: {
      description: {
        component:
          'Full screen loading component with Lottie animation for displaying loading states.',
      },
    },
  },
  decorators: [
    Story => (
      <View className="flex-1 bg-background-primary">
        <Story />
      </View>
    ),
  ],
  argTypes: {},
}

export default meta

type Story = StoryObj<typeof FullScreenLoading>

export const Default: Story = {
  render: () => <FullScreenLoading />,
  parameters: {
    docs: {
      description: {
        story: 'Default full screen loading with plane animation.',
      },
    },
  },
}

export const Centered: Story = {
  render: () => (
    <View className="flex-1 justify-center items-center">
      <FullScreenLoading />
    </View>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Full screen loading centered within a container.',
      },
    },
  },
}

export const WithCustomContainer: Story = {
  render: () => (
    <View className="flex-1 bg-background-secondary p-4 rounded-xl m-4">
      <FullScreenLoading />
    </View>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Full screen loading with custom container styling.',
      },
    },
  },
}

export const MultipleStates: Story = {
  render: () => (
    <View className="flex-1 gap-4">
      <View className="flex-1">
        <FullScreenLoading />
      </View>
      <View className="flex-1">
        <FullScreenLoading />
      </View>
    </View>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Multiple loading components with different background colors.',
      },
    },
  },
}

export const ThemeShowcase: Story = {
  render: () => (
    <View className="flex-1">
      <View className="flex-1 flex-row">
        <View className="flex-1 bg-background-quaternary">
          <View className="flex-1 bg-background-secondary m-2 rounded-lg">
            <FullScreenLoading />
          </View>
        </View>

        <View className="flex-1 bg-background-quaternary">
          <View className="flex-1 bg-background-secondary m-2 rounded-lg">
            <FullScreenLoading />
          </View>
        </View>
      </View>
    </View>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Loading component displayed in different theme contexts.',
      },
    },
  },
}

export const MultipleInstances: Story = {
  render: () => {
    const { selectedTheme } = useThemeStore()
    const colors = themeColors?.[selectedTheme]

    return (
      <View className="flex-1 flex-row flex-wrap">
        {Array.from({ length: 4 }, (_, index) => (
          <View
            key={index}
            className="w-1/2 h-1/2 border border-background-quaternary"
            style={{
              backgroundColor:
                index % 2 === 0 ? colors?.background?.primary : colors?.background?.secondary,
            }}
          >
            <FullScreenLoading />
          </View>
        ))}
      </View>
    )
  },
  parameters: {
    docs: {
      description: {
        story: 'Multiple loading instances to test performance and memory usage.',
      },
    },
  },
}

export const UsageExamples: Story = {
  render: () => {
    const { selectedTheme } = useThemeStore()
    const colors = themeColors?.[selectedTheme]

    const [showLoading, setShowLoading] = useState<boolean>(true)

    return (
      <View className="flex-1">
        <View className="p-4 bg-background-primary border-b border-background-quaternary">
          <TouchableOpacity
            activeOpacity={0.7}
            className="p-3 rounded-lg items-center"
            style={{
              backgroundColor: showLoading ? colors?.info : colors?.success,
            }}
            onPress={() => setShowLoading(!showLoading)}
          >
            <ThemedText type="body1" color="text-100">
              {showLoading ? 'Hide Loading' : 'Show Loading'}
            </ThemedText>
          </TouchableOpacity>
        </View>

        <View className="flex-1 relative">
          <View className="flex-1 p-4 justify-center items-center bg-background-primary">
            <View className="w-[200px] h-[200px] bg-background-quaternary rounded-xl mb-5" />
            <ThemedText type="body1" color="text-100" center>
              App Content Here
            </ThemedText>
          </View>

          {showLoading ? (
            <View className="absolute top-0 left-0 right-0 bottom-0 bg-background-secondary">
              <FullScreenLoading />
            </View>
          ) : null}
        </View>
      </View>
    )
  },
  parameters: {
    docs: {
      description: {
        story: 'Interactive example showing how to toggle loading state in a real app scenario.',
      },
    },
  },
}
