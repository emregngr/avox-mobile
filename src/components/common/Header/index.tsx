import { GlassView } from 'expo-glass-effect'
import * as Haptics from 'expo-haptics'
import { type ReactNode, useMemo } from 'react'
import type { ViewStyle } from 'react-native'
import { TouchableOpacity, View } from 'react-native'

import Back from '@/assets/icons/back.svg'
import { ThemedButtonText } from '@/components/common/ThemedButtonText'
import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import { cn } from '@/utils/common/cn'

type HeaderProps = {
  backIcon?: boolean
  backIconOnPress?: () => void
  containerClassName?: string
  hapticFeedback?: boolean
  isFavorite?: boolean
  rightButtonLabel?: string
  rightButtonOnPress?: () => void
  rightIcon?: ReactNode
  rightIconOnPress?: () => void
  shareIcon?: ReactNode
  shareIconOnPress?: () => void
  style?: ViewStyle
  testID?: string
  title?: string | string[]
  titleClassName?: string
}

export const Header = ({
  backIcon = true,
  backIconOnPress,
  containerClassName,
  hapticFeedback = false,
  isFavorite = false,
  rightButtonLabel,
  rightButtonOnPress,
  rightIcon,
  rightIconOnPress,
  shareIcon,
  shareIconOnPress,
  style,
  testID,
  title,
  titleClassName,
}: HeaderProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const STATIC_STYLES = {
    backIcon: {
      left: 16,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
    shareIcon: {
      right: 64,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
    rightIcon: {
      right: 16,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
  }

  const handleRightIconOnPress = () => {
    if (hapticFeedback) {
      if (isFavorite) {
        Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light)
      } else {
        Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Medium)
      }
    }

    rightIconOnPress?.()
  }

  return (
    <View
      className={cn('h-11 justify-center items-center', containerClassName)}
      style={style}
      testID={testID || 'header'}
    >
      {title ? (
        <ThemedText
          className={titleClassName ?? ''}
          color="text-100"
          ellipsizeMode="tail"
          numberOfLines={2}
          type="h3"
          center
        >
          {title}
        </ThemedText>
      ) : null}

      {backIcon ? (
        <GlassView
          style={STATIC_STYLES.backIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={20}
            onPress={backIconOnPress}
            testID="header-back-icon"
          >
            <Back color={colors?.onPrimary100} height={24} width={24} />
          </TouchableOpacity>
        </GlassView>
      ) : null}

      {shareIcon ? (
        <GlassView
          style={STATIC_STYLES.shareIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={shareIconOnPress}
            testID="header-share-icon"
          >
            {shareIcon}
          </TouchableOpacity>
        </GlassView>
      ) : null}

      {rightButtonLabel ? (
        <ThemedButtonText
          containerStyle="absolute right-4"
          label={rightButtonLabel}
          onPress={rightButtonOnPress as () => void}
          textColor="text-100"
          type="h4"
        />
      ) : rightIcon ? (
        <GlassView
          style={STATIC_STYLES.rightIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={handleRightIconOnPress}
            testID="header-right-icon"
          >
            {rightIcon}
          </TouchableOpacity>
        </GlassView>
      ) : null}
    </View>
  )
}
