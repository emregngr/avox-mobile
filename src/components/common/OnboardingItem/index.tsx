import { Image } from 'expo-image'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import type { OnBoardingType } from '@/types/common/onBoarding'

interface OnboardingItemProps {
  item: OnBoardingType
}

const STATIC_STYLES = {
  image: {
    height: '45%' as const,
    marginVertical: 20,
    width: '100%' as const,
  },
}

export const OnboardingItem = ({ item }: OnboardingItemProps) => {
  const { image, text, title } = item
  return (
    <View className="w-screen h-screen bg-background-primary">
      <Image
        cachePolicy="memory-disk"
        contentFit="contain"
        source={image}
        style={STATIC_STYLES.image}
        transition={0}
      />
      <View className="mx-4 z-10">
        <ThemedText
          className="mb-4"
          color="text-100"
          lineBreakMode="tail"
          numberOfLines={2}
          type="h1"
        >
          {title}
        </ThemedText>
        <ThemedText
          color="text-100" lineBreakMode="tail" numberOfLines={4}
          type="h3"
        >
          {text}
        </ThemedText>
      </View>
    </View>
  )
}
