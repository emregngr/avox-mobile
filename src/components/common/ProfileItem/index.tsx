import { MaterialCommunityIcons } from '@expo/vector-icons'
import type { ComponentType } from 'react'
import React, { useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'
import type { SvgProps } from 'react-native-svg'

import Right from '@/assets/icons/right.svg'
import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

type ProfileItemProps = {
  customLeftIcon?: ComponentType<SvgProps>
  danger?: boolean
  isLastItem?: boolean
  label: string
  leftIcon?: keyof typeof MaterialCommunityIcons.glyphMap
  onPress: () => void
  rightIcon?: boolean
  testID?: string
}

export const ProfileItem = ({
  customLeftIcon: CustomLeftIcon,
  danger,
  isLastItem = false,
  label,
  leftIcon,
  onPress,
  rightIcon = true,
  testID,
}: ProfileItemProps) => {
  const { selectedTheme } = useThemeStore()
  const colors = themeColors?.[selectedTheme]

  const iconColor = useMemo(
    () => (danger ? colors?.error : colors?.onPrimary100),
    [danger, colors?.error, colors?.onPrimary100],
  )

  const iconComponent = useMemo(() => {
    if (leftIcon) {
      return <MaterialCommunityIcons color={iconColor} name={leftIcon} size={24} />
    }
    if (CustomLeftIcon) {
      return <CustomLeftIcon color={iconColor} height={24} width={24} />
    }
    return null
  }, [leftIcon, CustomLeftIcon, iconColor])

  return (
    <>
      <TouchableOpacity
        activeOpacity={0.7}
        className="h-14 flex-row items-center justify-between px-4 bg-background-secondary"
        hitSlop={{ left: 20, right: 20 }}
        onPress={onPress}
        testID={testID || 'profile-item-touchable'}
      >
        <View className="flex-row items-center">
          {iconComponent}

          <ThemedText className="ml-4" color={danger ? 'error' : 'text-100'} type="body1">
            {label}
          </ThemedText>
        </View>

        {rightIcon ? (
          <Right
            color={iconColor} height={24} testID="right-icon"
            width={24}
          />
) : null}
      </TouchableOpacity>

      {!isLastItem ? (
        <View className="border-b border-background-quaternary ml-14" testID="separator" />
      ) : null}
    </>
  )
}
