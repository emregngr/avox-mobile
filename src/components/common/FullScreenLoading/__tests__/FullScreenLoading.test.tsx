import { render } from '@testing-library/react-native'
import React from 'react'

import { FullScreenLoading } from '@/components/common/FullScreenLoading'

describe('FullScreenLoading Component', () => {
  it('renders loading container and indicator', () => {
    const { getByTestId } = render(<FullScreenLoading />)

    expect(getByTestId('full-screen-loading')).toBeTruthy()
    expect(getByTestId('loading-indicator')).toBeTruthy()
  })

  it('has correct accessibility properties', () => {
    const { getByTestId } = render(<FullScreenLoading />)

    const loadingIndicator = getByTestId('loading-indicator')

    expect(loadingIndicator.props.size).toBe('large')
    expect(loadingIndicator.props.className).toBe('color-text-100')
  })

  it('container has correct styling', () => {
    const { getByTestId } = render(<FullScreenLoading />)

    const container = getByTestId('full-screen-loading')
    expect(container.props.className).toBe(
      'flex-1 items-center justify-center bg-background-primary',
    )
  })

  it('renders ActivityIndicator as loading state', () => {
    const { getByTestId } = render(<FullScreenLoading />)

    const indicator = getByTestId('loading-indicator')

    expect(indicator.props.animating).not.toBe(false)
  })
})

describe('FullScreenLoading Component Snapshot', () => {
  it('should render the FullScreenLoading Component successfully', () => {
    const { toJSON } = render(<FullScreenLoading />)

    expect(toJSON()).toMatchSnapshot()
  })
})
