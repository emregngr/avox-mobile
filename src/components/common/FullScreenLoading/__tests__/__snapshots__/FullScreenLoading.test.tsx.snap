// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FullScreenLoading Component Snapshot should render the FullScreenLoading Component successfully 1`] = `
<View
  className="flex-1 items-center justify-center bg-background-primary"
  testID="full-screen-loading"
>
  <ActivityIndicator
    className="color-text-100"
    size="large"
    testID="loading-indicator"
  />
</View>
`;
